pkg/apis/vsphere/v1/template_genclient.go

```diff
@@ -0,0 +1,97 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/url"
+	"path"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+var (
+	// ErrTemplateOperationNotSupported is returned if operations other than Get and List are performed
+	ErrTemplateOperationNotSupported = fmt.Errorf("%w: Template only support Get and List operations", api.ErrOperationNotSupported)
+)
+
+// EndpointURL returns the URL where to retrieve objects of type Template (only Get and List operations supported atm).
+func (t *Template) EndpointURL(ctx context.Context) (*url.URL, error) {
+	_, err := t.operationFromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	return url.ParseRequestURI(
+		fmt.Sprintf("/api/vsphere/v1/provisioning/templates.json/%s/%s?page=1&limit=1000", t.Location.Identifier, t.Type),
```

Comment:
Maybe we should return the URL without query parameters and add them in `FilterRequestURL`